stackslib/src/chainstate/nakamoto/tests/mod.rs

```diff
@@ -2839,3 +2847,304 @@ fn filter_one_transaction_per_signer_duplicate_nonces() {
+        let mut reward_set = RewardSet::empty();
+        reward_set.signers = Some(
+            signers
+                .iter()
+                .map(|(s, w)| {
+                    let mut signing_key = [0u8; 33];
+                    signing_key.copy_from_slice(
+                        &Secp256k1PublicKey::from_private(s)
+                            .to_bytes_compressed()
+                            .as_slice(),
+                    );
+                    NakamotoSignerEntry {
+                        signing_key,
+                        stacked_amt: 100_u128,
+                        weight: *w,
+                    }
+                })
+                .collect::<Vec<_>>(),
+        );
+        reward_set
+    }
+
+    #[test]
+    /// Base success case - 3 signers of equal weight, all signing the block
+    pub fn test_nakamoto_block_verify_signatures() {
+        let signers = vec![
+            Secp256k1PrivateKey::default(),
+            Secp256k1PrivateKey::default(),
+            Secp256k1PrivateKey::default(),
+        ];
+
+        let reward_set = make_reward_set(signers.iter().map(|s| (s.clone(), 100)).collect());
+
+        let mut header = NakamotoBlockHeader::empty();
+
+        // Sign the block sighash for each signer
+
+        let message = header.signer_signature_hash().0;
+        let signer_signature = signers
+            .iter()
+            .map(|s| s.sign(&message).expect("Failed to sign block sighash"))
+            .collect::<Vec<_>>();
+
+        header.signer_signature = signer_signature;
+
+        header
+            .verify_signer_signatures(&reward_set)
+            .expect("Failed to verify signatures");
+        // assert!(&header.verify_signer_signatures(&reward_set).is_ok());
+    }
+
+    #[test]
+    /// Fully signed block, but not in order
+    fn test_out_of_order_signer_signatures() {
+        let signers = vec![
+            (Secp256k1PrivateKey::default(), 100),
+            (Secp256k1PrivateKey::default(), 100),
+            (Secp256k1PrivateKey::default(), 100),
+        ];
+        let reward_set = make_reward_set(signers.clone());
+
+        let mut header = NakamotoBlockHeader::empty();
+
+        // Sign the block for each signer, but in reverse
+        let message = header.signer_signature_hash().0;
+        let signer_signature = signers
+            .iter()
+            .rev()
+            .map(|(s, _)| s.sign(&message).expect("Failed to sign block sighash"))
+            .collect::<Vec<_>>();
+
+        header.signer_signature = signer_signature;
+
+        match header.verify_signer_signatures(&reward_set) {
+            Ok(_) => panic!("Expected out of order signatures to fail"),
+            Err(ChainstateError::InvalidStacksBlock(msg)) => {
+                assert!(msg.contains("out of order"));
+            }
+            _ => panic!("Expected InvalidStacksBlock error"),
+        }
+    }
+
+    #[test]
+    // Test with 3 equal signers, and only two sign
```

Comment:
Once the math is corrected for calculating the threshold, you're going to want to test two edge cases with it:

* A signature fails because it has the largest possible weight that happens to be smaller than 70%
* A signature succeeds with _exactly_ 70%

Reply:
In fact, I'd recommend factoring the threshold calculation into its own method, so you can just test the math directly without having to construct reward sets for each test case.

Reply:
I've added those two scenarios and also some unit tests for the threshold calculation - feel free to resolve if you think they're covered 👍🏼